### ISSUE TITLE
No way to set cassandra configs (creds, tls option, etc) via docker-compose

### ISSUE THREAD

**Issue by markmark206** [temporalio/temporal#513]
(Discovered by Nicolas, thank you!)

## Expected Behavior

I should be able to specify cassandra configuration parameters (user name/ password, tls configuration options, replication factor) via parameters in my docker-compose file, similarly to how i can specify mysql user name and password here: https://github.com/temporalio/temporal/blob/16f23cb090955d405afdf6676881bdb6f434b043/docker/docker-compose-mysql.yml#L22-L23

## Actual Behavior

Unfortunately, there is no way to specify this for cassandra. The only thing that can be passed to the container is cassandra host(s) and keyspaces:

https://github.com/temporalio/temporal/blob/16f23cb090955d405afdf6676881bdb6f434b043/docker/docker-compose.yml#L13

It appears that our configuration template that we use when launching via docker-compose only expands CASSANDRA_SEEDS and *NAMESPACE values.

https://github.com/temporalio/temporal/blob/16f23cb090955d405afdf6676881bdb6f434b043/docker/config_template.yaml#L17-L23 

## Steps to Reproduce the Problem

  1. attempt to use docker compose to start temporal server running against a cassandra with credentials. fail miserably. ; ) 

## Specifications

  - Version: temporal 0.26.0
  - Platform: mac os

**Comment by mikhailshilkov** [comment]
That's a bummer! So is there a workaround to connect to a Cassandra cluster protected by credentials?

**Comment by gazmania** [comment]
Is there any movement on this? I've been trying to connect to Azure CosmosDB via the Cassandra API and can't get Temporal to recognise any options.  Then found this defect.

**Comment by mastermanu** [comment]
Hey folks,

We've updated the config_template file to take in a bunch of extra variables to pass in extra configuration information. The PR has been landed: https://github.com/temporalio/temporal/pull/869/files

Please take a look and let us know if this solves the issue, or if you have more questions!

**Comment by gazmania** [comment]
Hi.  I just pulled master to get this change.  Still not working for me.  I think the `start.sh` script needs updating to pass the additional info into the cqlsh for the `wait_for_cassandra()` function.  That's the first problem with trying to run from docker-compose for me with the new env vars, might be something else later on.

**Comment by mastermanu** [comment]
Will take another look at this today.

**Comment by mastermanu** [comment]
Ah, did not realize that you were using auto-setup image to run this (was assuming you were setting up Cassandra out of band and then trying to set environment variables on the server image).

This [PR](https://github.com/temporalio/temporal/pull/882) should partially address this as it will forward the relevant environment variables to the setup_cassandra_schema() function.

Will look at fixing the wait_for_cassandra() in a similar fashion tomorrow.

**Comment by mastermanu** [comment]
This [PR](https://github.com/temporalio/temporal/pull/885) should address the wait_for_cassandra() issue and has been merged. Are you able to verify that this unblocks you?

**Comment by mikhailshilkov** [comment]
@gazmania I tried connecting to my Cosmos DB with the latest Docker image but getting

```
gocql: unable to dial control conn 104.44.12.124: gocql: no response received from cassandra within timeout period
Unable to start server: cassandra schema version compatibility check failed: unable to create CQL Client: gocql: unable to create session: control: unable to connect to initial hosts: gocql: no response received from cassandra within timeout period.
```

Curious if you succeed with this. 

```
Environment:
      AUTO_SETUP:          true
      CASSANDRA_SEEDS:     mycassandra.cassandra.cosmos.azure.com
      CASSANDRA_USER:      mycassandra
      CASSANDRA_PORT:      10350
      CASSANDRA_PASSWORD:  blablamybse64pwd==
```

**Comment by gazmania** [comment]
@mikhailshilkov I haven't tried the latest version on master yet after the recent PRs.  When i was hacking about myself attempting to PoC connectivity to CosmosDB I managed to get the autosetup to work, sort of, after a couple of hit and miss runs.
* i saw issues with keyspace creation in that temporal-cassandra-tool returned without error but the ketspace didn't actually get created.  This happened inconsistently, sometimes it would work.  I assumed there was some sort of eventual consistency issue bu even after waiting a while the keyspace still didn't turn up.  I created manually if it didn't work.
* A few issues caused by CosmosDB implementation of Cassandra API where i had to tweak the schema
  * there is a hanging comma on the `serialised_event_batch` type which CosmosDB didn't like, Cassandra is evidently ok here
  * gc_grace_period has to be 0 for CosmosDB
  * named indexes are not supported in CosmosDB, have to remove the names all indexes

I was similar to the env vars you had above, but also was passing the below as I couldn't connect without some sort of TLS flags:
  "CASSANDRA_TLS_ENABLED=true"
  "SSL_VERSION=TLSv1_2"
  "SSL_VALIDATE=false"

On CosmosDB I can't find where to get the public cert so had the set SSL_VALIDATE to false.  This is what the helper instructions also do.

I haven't yet been able to actually get Temporal itself to work even after the schema was finally set up.  I'll let you know how that goes.

**Comment by mikhailshilkov** [comment]
@gazmania Thank you for diffing into that! I think it would make sense to open a new issue, summarize your findings, and eventually fix the auto-setup (in case Cosmos works as a backend at all).

**Comment by nhernandez2** [comment]
If it still works, what I had to do, in the case of cadence, was to use a custom config_template and modify the start.sh.

start.sh

`cqlsh -u $CASSANDRA_USER -p $CASSANDRA_PASSWORD --cqlversion=3.4.4 $server`

config_template

`default: 
cassandra:
hosts: {{ default .Env.CASSANDRA_SEEDS "" }}
keyspace: {{ default .Env.KEYSPACE "cadence" }}
user: {{ default .Env.CASSANDRA_USER "" }}
password: {{ default .Env.CASSANDRA_PASSWORD "" }}`

Dockerfile

`COPY config/config_template.yaml  /etc/cadence/config`
